### PR TITLE
CXEC-16839: Accessibility improvements for tab-bar and addtocart

### PR DESCRIPTION
**Pull request by steinsebastian**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts [hunk 1]
```diff
@@ -125,13 +125,22 @@ export class ConfiguratorStorefrontUtilsService {
...
   focusFirstAttribute(): void {
+    this.focusFirstActiveElement('cx-configurator-form');
+  }
+
+  /**
+   * Focus the first active element inside the given host element
+   *
+   * @param hostQuerySelector - query selector of the host element
+   */
+  focusFirstActiveElement(hostQuerySelector: string) {
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: Could you, please, rename `hostQuerySelector` parameter to `selector`?

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts [hunk 2]
```diff
@@ -125,13 +125,22 @@ export class ConfiguratorStorefrontUtilsService {
...
     if (!this.windowRef.isBrowser()) {
       return;
     }
-    const form = this.getElement('cx-configurator-form');
-    if (form) {
+    const tabBar = this.getElement(hostQuerySelector);
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: Could you, please, rename `tabBar` to `element`?

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts [hunk 3]
```diff
@@ -125,13 +125,22 @@ export class ConfiguratorStorefrontUtilsService {
    * Focus the first attribute in the form.
    */
   focusFirstAttribute(): void {
+    this.focusFirstActiveElement('cx-configurator-form');
+  }
+
+  /**
+   * Focus the first active element inside the given host element
+   *
+   * @param hostQuerySelector - query selector of the host element
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: Could you, please, rename `hostQuerySelector` parameter to `selector`?

File: feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts [hunk 4]
```diff
@@ -52,6 +67,92 @@ export class ConfiguratorTabBarComponent {
     )
   );
 
+  /**
+   * Navigates to the overview page and sets the focus on the overview element in the tab-bar.
+   * @param routerData - Router data
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts:73]: Please, add an empty line

File: feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts [hunk 5]
```diff
@@ -52,6 +67,92 @@ export class ConfiguratorTabBarComponent {
...
+        },
+      })
+      .then(() => {
+        this.focusOverviewInTabBar();
+      });
+  }
+
+  /**
+   * Navigates to the configuration page and sets the focus on the configuration element in the tab-bar.
+   * @param routerData - Router data
```
Comment (Larisa-Staroverova) [inline feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts:92]: Please, add an empty line

**Review by Larisa-Staroverova — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts (hunk 1 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: Done

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts (hunk 2 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: done

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts (hunk 3 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts]: done

File: feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts (hunk 4 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts:73]: done

File: feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts (hunk 5 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts:92]: done

**Review by Larisa-Staroverova — APPROVED** [review]

**Pull request merged**